### PR TITLE
FreeBSD fixes

### PR DESCRIPTION
**Pull request by notspiff**
Some issues encountered while building for FreeBSD. Still need to forward the C compiler to FFmpeg manually as discussed on slack but those maintaining those files should do that as they see fit.

### CODE REVIEW

File: lib/libUPnP/CMakeLists.txt [hunk 1]
```diff
@@ -96,9 +96,18 @@ else()
...
+                      Neptune/Source/System/Bsd/NptBsdSockets.cpp)
+endif()
+
 
 add_library(upnp STATIC ${SOURCES} ${HEADERS})
 target_compile_definitions(upnp PUBLIC -DNPT_CONFIG_ENABLE_LOGGING)
+if(CORE_SYSTEM_NAME STREQUAL "freebsd")
+  target_compile_definitions(upnp PUBLIC -DNPT_CONFIG_HAVE_GETADDRINFO)
+endif()
+                                       -DNPT_CONFIG_HAVE_GETADDRINFO)
```
Comment (Rechi) [inline lib/libUPnP/CMakeLists.txt]: mistake

**Comment by notspiff** [comment]
jenkins build this please

File: lib/libUPnP/CMakeLists.txt (hunk 1 above)
Comment (notspiff) [reply lib/libUPnP/CMakeLists.txt]: thx, forgot to commit the fix. rewritten.

**Comment by Rechi** [comment]
lib/libUPnP/CMakeLists.txt changes have to be added to lib/libUPnP/patches/. IMO you can just updated the existing patch

**Comment by notspiff** [comment]
why are we patching in things we authored? upstream only has scons.

**Comment by notspiff** [comment]
sillyness obeyed, patch updated.

**Comment by Rechi** [comment]
I don't know, @Montellese initially added the Makefile patch (autotools time). I only replaced it with a  CMakeLists patch.

**Review by wsnipex — APPROVED** [review]

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Pull request merged**